set-webpack-version.js

```diff
@@ -0,0 +1,25 @@
+/*
+ * This file sets Webpack version - both devDependencies and peerDependencies.
+ *
+ * This is, for some reason, needed with Windows and maybe npm 6. Running
+ * "npm install webpack -D" AND manually updating the peerDependencies
+ * (because npm 6 does not do that but npm 7 does) is not enough. For some
+ * reason why must, by hand, update the package.json file first and then
+ * run a normal "npm install".
+ */
```

Comment:
This is insanity, but THIS turned out to be the reason that `asset.info.sourceFilename` was missing on Windows (and why I could repeat it). I know from the Webpack folks - https://github.com/webpack/webpack/issues/12637#issuecomment-776185604 - that relying on `asset.info.sourceFilename` IS the correct way to do things.

Basically, for reasons that aren't clear, on Windows only, if you installed Webpack 5 via `npm install webpack -D` it would *not* work. Somewhere, internally, some dependencies must have gotten "crossed". But if you manually change the `peerDependencies` and `devDependencies` for Webpack and then run `npm install`, things work fine. This... almost killed me. But it turns out that this was an edge-case npm + Windows bug of some sort... or possibly some edge case since the `package.json` file is mixed with libraries that have Webpack 4 vs Webpack 5 as their peer dependency.

Reply:
This is fascinating. What we need to know before we merge is why this was an issue only _now_. We've been testing webpack v4 and webpack v5 separately with the windows workflows without issue up to this point. So what's the missing bit of nuance as to why it's an issue now?

Reply:
@weaverryan ping on this comment